photon-lib/src/main/scala/com/linkedin/photon/ml/data/MinHeapWithFixedCapacity.scala

```diff
@@ -62,67 +91,89 @@ protected[ml] class MinHeapWithFixedCapacity[T <: Comparable[T] : ClassTag](capa
    * @return The updated heap
    */
   def ++=(minHeapWithFixedCapacity: MinHeapWithFixedCapacity[T]): this.type = {
-    if (cumCount + minHeapWithFixedCapacity.cumCount < capacity) {
-      arrayBuffer.addAll(minHeapWithFixedCapacity.arrayBuffer)
+
+    cumCount += minHeapWithFixedCapacity.cumCount
```

Comment:
This implementation is not thread-safe. We should probably put a warning in the top docblock.

Reply:
Will do - though it is safe in our use case as heaps are only merged once all insertions are completed.